src/cursor_manager.ts

```diff
@@ -361,14 +360,24 @@ export class CursorManager implements Disposable {
                 }], isMultiSelection: ${editor.selections.length > 1}`,
             );
 
-            if (selection.isEmpty) {
-                // exit visual mode when clicking elsewhere
-                if (this.main.modeManager.isVisualMode && kind == TextEditorSelectionChangeKind.Mouse)
-                    await this.client.input("<Esc>");
+            if (this.main.modeManager.isInsertMode) {
+                // Keep the cursor consistent as much as possible to avoid
+                // changes in editor selections when synchronizing the cursor
+                // position of neovim to vscode in insert mode, thus breaking
+                // the cursor-related functions of vscode, such as placeholder
+                // jumps in snippet mode, multi-cursor selection, etc.
+                this.setWantInsertCursorUpdate(editor, false);
                 await this.updateNeovimCursorPosition(editor, selection.active);
             } else {
-                if (kind != TextEditorSelectionChangeKind.Mouse || !config.disableMouseSelection)
-                    await this.updateNeovimVisualSelection(editor, selection);
+                if (selection.isEmpty) {
+                    // exit visual mode when clicking elsewhere
+                    if (this.main.modeManager.isVisualMode && kind == TextEditorSelectionChangeKind.Mouse)
+                        await this.client.input("<Esc>");
+                    await this.updateNeovimCursorPosition(editor, selection.active);
+                } else {
+                    if (kind != TextEditorSelectionChangeKind.Mouse || !config.disableMouseSelection)
```

Comment:
I'm surprised the linter didn't flag this usage of `!=` vs `!==`

Reply:
need to enable "eqeqeq" : https://github.com/neovim/node-client/blob/8982da5532278f178286a11696939cbf241b4046/.eslintrc.js#L37-L38

Reply:
Yeah, I assumed (incorrectly) it was in the recommended ruleset :)

Reply:
`npx eslint-flat-config-viewer` can see where the problem is